### DISCUSSION TITLE
Why rollup() method definition in [@types/d3-array] doesn't have keys as array spread?

### DISCUSSION THREAD

**Discussion by KupnoH** [DefinitelyTyped/DefinitelyTyped#57356] [Issues with a @types package] [answered]
Hi.
I've been using @types/d3-array (v3.0.2) and noticed that rollup method accepts only 3-5 arguments, though in d3-array js package it can have more than 5 arguments because it's definition has array spreading for keys (3rd - 5th attribute here) so in original js package I can use 4 or more keys to use rollup method, but due to this definition I can use only 3 keys to rollup.

Here's definition of rollup from [@types/d3-array](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/54d540ab4deb2588c0eff39dadf370cbf0a2dee4/types/d3-array/index.d.ts#L643):
```
/**
 * Groups and reduces the specified array of values into an InternMap from key to value.
 *
 * @param iterable The array to group.
 * @param reduce The reduce function.
 * @param key1 The first key function.
 * @param key2 The second key function.
 * @param key3 The third key function.
 */
export function rollup<TObject, TReduce, TKey1, TKey2, TKey3>(
    iterable: Iterable<TObject>,
    reduce: (value: TObject[]) => TReduce,
    key1: (value: TObject) => TKey1,
    key2: (value: TObject) => TKey2,
    key3: (value: TObject) => TKey3
): InternMap<TKey1, InternMap<TKey2, InternMap<TKey3, TReduce>>>;
```

and here's definition from [original d3-array package](https://github.com/d3/d3-array/blob/894a718fa126ad530a8523b4d12f2656d8fbb44e/src/group.js#L27):

```
export function rollup(values, reduce, ...keys) {
  return nest(values, identity, reduce, keys);
}
```

So is rollup definition correct? And if so is there any workaround for the case when I need more than 3 keys?

Thanks in advance,
Best regards

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment, marked as answer]
I believe those types were written before TypeScript had more capabilities with generic spread types. Feel free to contribute a PR to expand the types to handle an unlimited number of keys!

**Reply by KupnoH** [reply]
Great idea! Thanks a lot for a great tip :)
